components/mpas-albany-landice/driver/glc_comp_mct.F

```diff
@@ -1385,49 +1390,78 @@ subroutine glc_import_mct(x2g_g, errorCode)
+   real(kind=RKIND), pointer :: config_ice_density
+   real(kind=RKIND), pointer :: config_invalid_value_TF
+   real(kind=RKIND) :: fractionalMaskVal
 
     errorCode = 0
 
+    call mpas_pool_get_config(domain % configs, 'config_ice_density', config_ice_density)
+    call mpas_pool_get_config(domain % configs, 'config_invalid_value_TF', config_invalid_value_TF)
+
     n = 0
     block => domain % blocklist
 
     do while (associated(block))
          call mpas_pool_get_subpool(block % structs, 'mesh', meshPool)
-         call mpas_pool_get_dimension(meshPool, 'nCellsSolve', nCellsSolve)
-
-         ! Get variables from pools
          call mpas_pool_get_subpool(block % structs, 'geometry', geometryPool)
          call mpas_pool_get_subpool(block % structs, 'thermal', thermalPool)
+         call mpas_pool_get_subpool(block % structs, 'extrapOceanData', extrapOceanDataPool)
 
+         ! Get dimensions
+         call mpas_pool_get_dimension(meshPool, 'nCellsSolve', nCellsSolve)
+         call mpas_pool_get_dimension(geometryPool, 'nISMIP6OceanLayers', nISMIP6OceanLayers)
+
+         ! Get variables from pools
          call mpas_pool_get_array(geometryPool, 'sfcMassBal', sfcMassBal)
          call mpas_pool_get_array(geometryPool, 'floatingBasalMassBal',floatingBasalMassBal)
          call mpas_pool_get_array(thermalPool, 'surfaceTemperature',surfaceTemperature)
-         call mpas_pool_get_array(geometryPool, 'ismip6_2dThermalForcing', ismip6_2dThermalForcing)
+         call mpas_pool_get_array(geometryPool, 'ismip6shelfMelt_3dThermalForcing', ismip6shelfMelt_3dThermalForcing)
+         call mpas_pool_get_array(extrapOceanDataPool, 'orig3dOceanMask', orig3dOceanMask)
 !         call mpas_pool_get_array(thermalPool, 'basalOceanHeatflx',basalOceanHeatflx)
          !call mpas_pool_get_array(geometryPool, 'OceanDensity',OceanDensity)
 
+         if (nISMIP6OceanLayers > 0) then
+            orig3dOceanMask(:,:) = 0
+            ismip6shelfMelt_3dThermalForcing(:,:) = 0.0_RKIND
+         endif
          do i = 1, nCellsSolve
             n = n + 1
             sfcMassBal(i) = x2g_g % rAttr(index_x2g_Flgl_qice, n)
             floatingBasalMassBal(i) = x2g_g % rAttr(index_x2g_Fogx_qiceli, n)
-            if (ocn_c2_glctf) &
-               ismip6_2dThermalForcing(i) = x2g_g % rAttr(index_x2g_So_tf2d, n)
+            if (nISMIP6OceanLayers > 0) then
+               do iLev = 1, nISMIP6OceanLayers
+                  fractionalMaskVal = x2g_g % rAttr(index_x2g_So_tf3d_mask(iLev), n)
+                  if (fractionalMaskVal > 0.5_RKIND) then
+                     ! Only use MALI grid cells that have at least 50% overlap with MPAS-Ocean cells
+                     ! if valid, mark our integer mask as 1 and scale the TF value by the fractional mask
+                     orig3dOceanMask(iLev,i) = 1
+                     ismip6shelfMelt_3dThermalForcing(iLev, i) = x2g_g % rAttr(index_x2g_So_tf3d(iLev), n) / fractionalMaskVal
```

Comment:
Is this a way to scale melt rates, since we can assume that melt rates are linearly related to Tf?

Reply:
Oh, I was assuming this was a spatial fraction but is it actually a time fraction?